.github/workflows/python-poetry-release.yaml

```diff
@@ -0,0 +1,98 @@
+name: Reusable workflow for building and publishing python packages to TestPyPI or PyPI
+
+on:
+  workflow_call:
+    inputs:
+      release-type:
+        description: "Scope of the release; See: https://python-poetry.org/docs/cli/#version"
+        required: true
+        type: string
+      ref:
+        description: "The ref name to checkout the repository."
+        required: false
+        default: ${{ github.event.repository.default_branch }}
+        type: string
+      python-version:
+        description: "The python version for setting up poetry. (Default is 3.7)"
+        required: false
+        default: "3.7"
+        type: string
+      poetry-version:
+        description: "The python version for setting up poetry. (Default is 1.1.12)"
+        required: false
+        default: "1.1.12"
+        type: string
+      publish-to-test:
+        description: "If set to false, the packages are published to pypi.org. (Default is true)"
+        required: false
+        default: true
+        type: boolean
+      working-directory:
+        description: "The working directory of your Python package. (Default is root directory)"
+        required: false
+        default: "./"
+        type: string
+    secrets:
+      github-username:
+        description: "The github username for pushing."
+        required: true
+      github-email:
+        description: "The github email for pushing."
```

Comment:
nitpick: isn't it rather used for the commit?

Reply:
True. Let's change this to: `The GitHub Email for committing the changes`. WDYT?